src/org/graphstream/algorithm/LongestPath.java

```diff
@@ -0,0 +1,144 @@
+package org.graphstream.algorithm;
+
+import org.graphstream.graph.Edge;
+import org.graphstream.graph.Graph;
+import org.graphstream.graph.Node;
+import org.graphstream.graph.implementations.Graphs;
+
+import java.util.*;
+
+/**
+ * Implementation of longest path algorithm in a DAG (directed acyclic graph) using topologicalSort
+ *
+ * @complexity yet to be calculated
+ */
+public class LongestPath implements Algorithm {
+
+    /**
+     * graph to calculate longest path
+     */
+    private Graph graph;
+
+    /**
+     * map with all disctances from starting point
+     */
+    private Map<Node, Double> distanceMap;
+
+    /**
+     * calculated longest path
+     */
+    private List<Node> longestPath;
+
+    /**
+     * node and value at the end of the longest path
+     */
+    private Map.Entry<Node, Double> longestPathNode;
+
+    /**
+     * weighted or unweighted graph
+     */
+    private boolean weighted = true;
+
+    public void init(Graph theGraph) {
+        graph = Graphs.clone(theGraph);
```

Comment:
It seems not really necessary.

Reply:
@hichbra do you mean the graph cloning or the weighted attribute?

Reply:
The graph cloning

Reply:
yes, the cloning is not necessary because the algorithm is not changing the graph, just saw that all other algorithms are doing this. I will remove this...

Reply:
What do you think about that @pigne ? All algorithm should be use a copy?